gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/op/AbstractEvalOpProcessor.java

```diff
@@ -191,45 +193,21 @@ protected AbstractEvalOpProcessor(final boolean manageTransactions) {
      * A generalized implementation of the "eval" operation.  It handles script evaluation and iteration of results
      * so as to write {@link ResponseMessage} objects down the Netty pipeline.  It also handles script timeouts,
      * iteration timeouts, metrics and building bindings.  Note that result iteration is delegated to the
-     * {@link #handleIterator} method, so those extending this class could override that method for better control
-     * over result iteration.
-     *
-     * @param context The current Gremlin Server {@link Context}
+     * {@link #handleIterator(Context, Iterator)} method, so those extending this class could override that method for
+     * better control over result iteration.
+     *  @param ctx The current Gremlin Server {@link Context}. This handler ensures that only a single final
+     *            response is sent to the client.
      * @param gremlinExecutorSupplier A function that returns the {@link GremlinExecutor} to use in executing the
      *                                script evaluation.
      * @param bindingsSupplier A function that returns the {@link Bindings} to provide to the
-     *                         {@link GremlinExecutor#eval} method.
-     * @see #evalOpInternal(ResponseHandlerContext, Supplier, BindingSupplier)
+ *                         {@link GremlinExecutor#eval} method.
      */
-    protected void evalOpInternal(final Context context, final Supplier<GremlinExecutor> gremlinExecutorSupplier,
+    protected void evalOpInternal(final Context ctx, final Supplier<GremlinExecutor> gremlinExecutorSupplier,
                                   final BindingSupplier bindingsSupplier) throws OpProcessorException {
-        final ResponseHandlerContext rhc = new ResponseHandlerContext(context);
-        try {
-            evalOpInternal(rhc, gremlinExecutorSupplier, bindingsSupplier);
-        } catch (Exception ex) {
-            // Exceptions may occur on after the script started executing, therefore corresponding errors must be
-            // reported via the ResponseHandlerContext.
-            logger.warn("Unable to process script evaluation request: " + ex, ex);
-            rhc.writeAndFlush(ResponseMessage.build(context.getRequestMessage())
-                    .code(ResponseStatusCode.SERVER_ERROR)
-                    .statusAttributeException(ex)
-                    .statusMessage(ex.getMessage()).create());
-        }
-    }
-
-    /**
-     * A variant of {@link #evalOpInternal(Context, Supplier, BindingSupplier)} that is suitable for use in situations
-     * when multiple threads may produce {@link ResponseStatusCode#isFinalResponse() final} response messages
-     * concurrently.
-     * @see #evalOpInternal(Context, Supplier, BindingSupplier)
-     */
-    protected void evalOpInternal(final ResponseHandlerContext rhc, final Supplier<GremlinExecutor> gremlinExecutorSupplier,
```

Comment:
We are keeping `ResponseHandlerContext` as a deprecated class, but we are removing this `evalOpInternal` overload? If we intend to keep `ResponseHandlerContext` for backward compatibility would we also need to keep the methods that use it as parameters?

Sorry, I did not review very closely, perhaps I missed something in this regard.

Reply:
....what's going on here?!  i fixed all that! that was the whole pont of my change. seems like my PR is messed up somehow. looking

Reply:
ok - i don't know what happened, but somehow that change got reverted somewhere along the line. thanks for noticing that @dimas-b -